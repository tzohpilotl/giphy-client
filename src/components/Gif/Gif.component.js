import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { isFavorite } from '../../store/selectors';
import { toggleFavorite } from '../../store/favorites.reducer';
import HeartFilled from './heart-filled.png';
import Heart from './heart.png';
import './Gif.css';

export const Gif = ({ id, isFavorite, gif, toggleFavorite }) => {
  const title = gif.title;
  const imageUrl = gif.images.original.url;

  return (
    <div className="image-section" onClick={() => toggleFavorite(id, gif)}>
      <div className="favorite-icon">
        <img src={isFavorite ? HeartFilled : Heart} alt="Heart" />
      </div>
      <img src={imageUrl} alt={title} />
    </div>
  );
};

const mapStateToProps = (state, ownProps) => ({
  isFavorite: isFavorite(state, ownProps.id),
});

const mapDispatchToProps = dispatch => ({
  toggleFavorite: bindActionCreators(toggleFavorite, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Gif);
