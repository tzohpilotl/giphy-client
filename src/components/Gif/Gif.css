.image-section {
  position: relative;
}

.favorite-icon {
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
}