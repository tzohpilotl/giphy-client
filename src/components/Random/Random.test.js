import React from 'react';
import { shallow, mount } from 'enzyme';
import RandomContainer, { Random } from './Random.component';
import Gif from '../Gif';
import Spinner from '../Spinner';
import Button from '../Button';
import { getRandomGif } from '../../api';

jest.mock('../../api', () => ({ getRandomGif: jest.fn() }));

const randomGifData = {
  data: {
    title: 'My gif',
    images: {
      original: { url: 'http://thegifurl.com' },
    },
  },
};

describe('Random component', () => {
  it('should render', () => {
    expect(shallow(<Random />).isEmptyRender()).toEqual(false);
  });

  it('should match snapshots', () => {
    expect(shallow(<Random />)).toMatchSnapshot();
  });

  it('should fetch a new gif on mounting', () => {
    mount(<RandomContainer />);
    expect(getRandomGif).toHaveBeenCalled();
  });

  it('should load the spinner if no gif is loaded', () => {
    const wrRandomer = shallow(<Random />);
    expect(wrRandomer.containsMatchingElement(<Spinner />)).toBe(true);
  });

  it('should render a gif if the component has the gif info', () => {
    const wrRandomer = shallow(<Random gif={randomGifData} />);
    const gif = wrRandomer.find(Gif);
    expect(gif).toHaveLength(1);
  });

  it('should fetch a new gif if the button is clicked', () => {
    const wrRandomer = shallow(<Random />);
    wrRandomer.find(Button).simulate('click');
    expect(getRandomGif).toHaveBeenCalled();
  });
});
