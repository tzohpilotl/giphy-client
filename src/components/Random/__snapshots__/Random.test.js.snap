// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random component should match snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Random />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(button)
          disabled={true}
        >
          show me another one
        </Styled(button)>,
        <Spinner />,
      ],
      "className": "Random",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "show me another one",
          "disabled": true,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "show me another one",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(button)
            disabled={true}
          >
            show me another one
          </Styled(button)>,
          <Spinner />,
        ],
        "className": "Random",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "show me another one",
            "disabled": true,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "show me another one",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
